stress_test/controller.py

```diff
@@ -0,0 +1,349 @@
+# Copyright (c) 2016 Intracom S.A. Telecom Solutions. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v1.0 which accompanies this distribution,
+# and is available at http://www.eclipse.org/legal/epl-v10.html
+
+""" Controller Class- All controller-related functionality is here"""
+
+#import collections
+import json
+import common
+import logging
+import os
+import subprocess
+import time
+import util.netutil
+import util.process
+
+
+
+class Controller:
+
+
+    def __init__(self,ctrl_base_dir,test_config):
+
+#        logging.info('[Controller Class] Parsing test configuration')
+#        with open(json_file) as json_conf_file:
+#        self.test_config = json.load(json_conf_file)
+
+        """Create a Controller object. Options from JSON input file
+        :param test_config: JSON input configuration
+        :type test_config: JSON configuration dictionary
+        """
+        self.__dict__ = test_config
+        self.name = test_config['controller_name']
+        logging.info('Initializing parameters for the {0} Controller '.format(self.name))
+
+        #self.host_ip = test_config['controller_host_ip']
+        self.node_ip = test_config['controller_node_ip']
+        self.node_ssh_port = test_config['controller_node_ssh_port']
+        self.node_username = test_config['controller_node_username']
+        self.node_password = test_config['controller_password']
+
+        self.logs_dir = ctrl_base_dir + test_config['controller_logs_dir']
+        self.rebuild = ctrl_base_dir + test_config['controller_rebuild']
+        self.cleanup = ctrl_base_dir + test_config['controller_cleanup']
+        self.port = test_config['controller_port']
+        self.cpu_shares = test_config['controller_cpu_shares']
+        self.logs_dir = ctrl_base_dir + test_config['controller_logs_dir']
+
+        self.build_handler = ctrl_base_dir + test_config['controller_build_handler']
+        self.start_handler = ctrl_base_dir + test_config['controller_start_handler']
+        self.stop_handler = ctrl_base_dir + test_config['controller_stop_handler']
+        self.status_handler = ctrl_base_dir + test_config['controller_status_handler']
+        self.clean_handler = ctrl_base_dir + test_config['controller_clean_handler']
+
+
+        self.java_opts = ' '.join(test_config['java_opts'])
+        self.cpid = -1
+
+        if 'controller_restconf_port' in test_config:
+           self.restconf_port = test_config['controller_restconf_port']
+           self.restconf_user = test_config['controller_restconf_user']
+           self.restconf_password = test_config['controller_restconf_password']
+
+    def cleanup(self,ssh_client=None):
```

Comment:
comment for all member functions: `ssh_client` will not be needed as an argument anymore, since it will be a class attribute initialized once (via `init_ssh` method)